src/js/background/context-menu.js

```diff
@@ -0,0 +1,388 @@
+  },
+  listeners: {
+    onFillInAddressWithAliasId: async (info, tab) => {
+      // Trim the context menu id to get the alias reference ID
+      const selectedAliasId = info.menuItemId.replace(
+        "fx-private-relay-use-existing-alias_",
+        ""
+      );
+
+      // Get stored alias data
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+
+      // Select the correct alias from the stored alias data
+      const selectedAliasObject = relayAddresses.filter((alias) => {
+        return alias.id === parseInt(selectedAliasId, 10);
+      });
+
+      browser.tabs.sendMessage(
+        tab.id,
+        {
+          type: "fillTargetWithRelayAddress",
+          targetElementId: info.targetElementId,
+          relayAddress: selectedAliasObject[0],
+        },
+        {
+          frameId: info.frameId,
+        }
+      );
+      
+    },
+    onLocalStorageChange: async (changes, area) => {
+      let changedItems = Object.keys(changes);
+      for (let item of changedItems) {
+        if (item === "relayAddresses") {
+          await relayContextMenus.init();
+        }
+
+        if (item === "apiToken" && changes[item].newValue === undefined) {
+          // User has logged out. Remove all menu items.
+          await browser.menus.removeAll();
+        }
+      }
+    },    
+    onMakeRelayAddressForTargetElement: async (info, tab) => {
+      const pageUrl = new URL(info.pageUrl);
+      const newRelayAddress = await makeRelayAddress(pageUrl.hostname);
+    
+      if (newRelayAddress.status === 402) {
+        browser.tabs.sendMessage(tab.id, {
+          type: "showMaxNumAliasesMessage",
+        });
+        return;
+      }
+    
+      await browser.tabs.sendMessage(
+        tab.id,
+        {
+          type: "fillTargetWithRelayAddress",
+          targetElementId: info.targetElementId,
+          relayAddress: newRelayAddress,
+        },
+        {
+          frameId: info.frameId,
+        }
+      );
+    }
+  },
+  menus: {
+    create: async (data, opts=null) => {     
+      // If multiple items need to be created: 
+      if (Array.isArray(data)) {
+        data.forEach(async (menu) => {
+          await browser.menus.create(menu, relayContextMenus.utils.onCreatedCallback);
+        });
+
+        return;
+      }
+
+      // Loop Through Existing Aliases
+      if (opts?.createExistingAliases) {
+        
+        // TODO: Edgecase Fix if API doesn't respond (use local)
+        const aliases = await relayContextMenus.utils.getAliases();
+        
+        const filteredAliases = opts.currentWebsite
+          ? relayContextMenus.utils.getSiteSpecificAliases(
+              aliases,
+              opts.currentWebsite
+            )
+          : relayContextMenus.utils.getMostRecentAliases(aliases);
+
+        // // Cache server set locally
+        // browser.storage.local.set({ relayAddresses: aliases });
```

Comment:
```suggestion
```

Reply:
Removed in 5cf283312e7756533cd32b3be3a0db633577a86d